internal/step/foreach/provider.go

```diff
@@ -483,23 +487,39 @@ func (r *runningStep) ProvideStageInput(stage string, input map[string]any) erro
 	case string(StageIDExecute):
 		items := input["items"]
 		v := reflect.ValueOf(items)
-		input := make([]any, v.Len())
+		subworkflowInputs := make([]any, v.Len())
 		for i := 0; i < v.Len(); i++ {
 			item := v.Index(i).Interface()
 			_, err := r.workflow.Input().Unserialize(item)
 			if err != nil {
 				return fmt.Errorf("invalid input item %d for subworkflow (%w) for run/step %s", i, err, r.runID)
 			}
-			input[i] = item
+			subworkflowInputs[i] = item
 		}
 		if r.executionInputAvailable {
 			return fmt.Errorf("input for execute workflow provided twice for run/step %s", r.runID)
 		}
+		parallelismInput := input["parallelism"]
+		var parallelism int64
+		if parallelismInput != nil {
+			serializedParallelismInput, err := parallelismSchema.Unserialize(parallelismInput)
+			if err != nil {
+				return fmt.Errorf("failed to unserialized parallelism input for run/step %s: %w", r.runID, err)
+			}
+			parallelism = serializedParallelismInput.(int64)
+		} else {
+			parallelism = int64(1)
+		}
```

Comment:
Is line 511 doing the right thing?  I think the schema is already providing a default value, so we probably never get to line 511; however, if we were for some reason to get there, shouldn't it be providing the value from the schema rather than a hard-coded value (which, at the moment, happens to be the same as the one in the schema, but if the schema-default were to be changed in the future, will anyone know to update the value here??)?

Reply:
That does not work for some reason. Maybe we need to improve our handling of defaults.

```
Bug: failed to provide input to step subwf_step (%!w(*fmt.wrapError=&{failed to unserialized parallelism input for run/step subwf_step: <nil> cannot be converted to an int64 0x14000483ea0}))
```